### ISSUE TITLE
Weird behaviour with nested repeated fields

### ISSUE THREAD

**Issue by Allsimon** [mpartel/postgres-protobuf#3]
Hello,

I get wrong results when fetching nested repeated fields. I'm not sure what I missed since this use case seems to be tested [here](https://github.com/mpartel/postgres-protobuf/blob/master/expected/postgres_protobuf.out#L252)
```
SELECT protobuf_query_multi(
  'pgpb.test.ExampleMessage:repeated_inner[*].inner_repeated[*]',
  '\x2a0a120361626312036465662a0a12036161611203626262'::BYTEA
) AS result;
result
abc
def
aaa
bbb
(4 rows)
```

I created a repo in order to reproduce it: https://github.com/Allsimon/postgres-protobuf/blob/bug/reproduction/bug/README.md

**Comment by mpartel** [comment]
Thanks for the report and the convenient repro!

The "Second price is missing" case happens because you use `protobuf_query` instead of `protobuf_query_multi` or `protobuf_query_array`. I've added emphasis to the readme that `protobuf_query` only returns the first result.

The index case looks broken but I don't know why yet. I haven't managed to repro it in unit tests yet. I'll keep trying.

**Comment by mpartel** [comment]
Ok, should be fixed now with v0.2.2!

**Comment by Allsimon** [comment]
That was fast, thanks a lot !
